src/safeds/exceptions/_generic.py

```diff
@@ -0,0 +1,166 @@
+from __future__ import annotations
+
+from abc import ABC, abstractmethod
+
+
+class OutOfBoundsError(ValueError):
+    """A generic exception that can be used to signal that a (float) value is outside its expected range."""
+
+    def __init__(self, actual: float, *, lower_bound: Bound | None = None, upper_bound: Bound | None = None):
+        """
+        Initialize an OutOfBoundsError.
+
+        Parameters
+        ----------
+        actual: float
+            The actual value that is outside its expected range.
+        lower_bound: Bound | None
+            The lower bound of the expected range.
+        upper_bound: Bound | None
+            The upper bound of the expected range.
+
+        Raises
+        ------
+        ValueError
+            If upper_bound < lower_bound or if actual does not lie outside the given interval.
+        """
```

Comment:
According to the[ `numpydoc` style guide](https://numpydoc.readthedocs.io/en/latest/format.html#class-docstring), parameters of the `__init__` method should be documented in the docstring of the _class_.

It's OK, though, to _repeat_ it in the docstring of the `__init__`, so they also show up in the rendered documentation for `__init__`.

Reply:
Done.